pkg/api/plugins.go

```diff
@@ -513,6 +507,50 @@ func (hs *HTTPServer) pluginMarkdown(ctx context.Context, pluginID string, name
 	return md.Content, nil
 }
 
+// hasPluginRequestedPermissions verifies an installer has the permissions that the plugin requests to have on Grafana.
+func (hs *HTTPServer) hasPluginRequestedPermissions(c *contextmodel.ReqContext, pluginID, version string) error {
+	repoCompatOpts := repo.NewCompatOpts(hs.Cfg.BuildVersion, runtime.GOOS, runtime.GOARCH)
+
+	jsonData, err := hs.pluginRepo.GetPluginJson(c.Req.Context(), pluginID, version, repoCompatOpts)
+	if err != nil {
+		return err
+	}
+
+	// No registration => Early return
+	if jsonData.ExternalServiceRegistration == nil || len(jsonData.ExternalServiceRegistration.Permissions) == 0 {
+		return nil
+	}
+
+	hs.log.Debug("check installer's permissions, plugin wants to register an external service")
+	evaluator := evalAllPermissions(jsonData.ExternalServiceRegistration.Permissions)
+	hasAccess := accesscontrol.HasAccess(hs.AccessControl, c)
+	if !hs.Cfg.RBACSingleOrganization {
+		// If this is not in a single organization setup, the installer needs to have plugin's permissions accross all orgs
+		// TODO (gamab) this is probably going to fail for a lot of permissions since we only grant server admin permissions globally.
```

Comment:
@IevaVasiljeva @kalleep I'd be interested to have your thoughts on this :thinking: